app/code/Magento/QuoteGraphQl/Model/Cart/AddProductsToCart.php

```diff
@@ -50,16 +50,16 @@ public function __construct(
      */
     public function execute(Quote $cart, array $cartItems): void
     {
-        foreach ($cartItems as $cartItemData) {
```

Comment:
We can't move this logic after error checking.

Now it works as:
1) Perform adding products to cart
2) Check errors after the previous operation

Reply:
changes were reverted